SpinCore_pp/save_data.py

```diff
@@ -0,0 +1,76 @@
+import h5py
+import os
+import pyspecdata as psd
+import pyspecProcScripts
+import subprocess
+
+
+def save_data(dataset, target_directory, config_dict, counter_type):
+    """save data to an h5 file with appropriately labeled nodename and performs
+    rough processing
+
+    Parameters
+    ==========
+    dataset: nddata
+        acquired data in nddata format
+    my_exp_type: str
+        directory on the share drive you want to save to
+    config_dict: dict
+        config_dict pulled from the active.ini file
+    counter_type: str
+        type of counter you are incrementing
+
+    Returns
+    =======
+    config_dict: dict
+        the updated config dict after appropriately incrementing the counter
+    """
+    # {{{ create filename
+    filename = f"{config_dict['date']}_{config_dict['chemical']}_{config_dict['type']}"
+    # }}}
+    nodename = config_dict["type"] + "_" + str(config_dict["%s_counter" % counter_type])
+    dataset.name(nodename)
+    filename_out = filename + ".h5"
+    if os.path.exists(f"{target_directory}{filename_out}"):
+        print("this file already exists so we will add a node to it!")
+        with h5py.File(
+            os.path.normpath(os.path.join(target_directory, f"{filename_out}"))
+        ) as fp:
+            while nodename in fp.keys():
+                config_dict["%s_counter" % counter_type] += 1
+                nodename = (
+                    config_dict["type"]
+                    + "_"
+                    + str(config_dict["%s_counter" % counter_type])
+                )
+            dataset.name(nodename)
+    dataset.hdf5_write(f"{filename_out}", directory=target_directory)
+    print("\n** FILE SAVED IN TARGET DIRECTORY ***\n")
+    target_directory_parts = target_directory.split("\\")
```

Comment:
this doesn't work across operating systems.  Use `os.path.split`

Reply:
but see the comment below about exp_type, which should make this not necessary anyways

Reply:
fixed and takes my_exp_type as kwarg now